pkg/core/blockchain.go

```diff
@@ -754,49 +767,71 @@ func (bc *Blockchain) resetStateInternal(height uint32, stage stateChangeStage)
 
 		const persistBatchSize = 10000
 		var (
-			seekErr error
-			cnt     int
+			seekErr        error
+			cnt            int
+			storageItmsCnt int
+			contractIDsCnt int
+			batchCnt       int
 		)
 		trieStore.Seek(storage.SeekRange{Prefix: []byte{byte(oldStoragePrefix)}}, func(k, v []byte) bool {
+			if seekErr != nil {
+				return false
+			}
 			if cnt >= persistBatchSize {
-				_, seekErr = cache.Persist()
+				cnt = 0
+				keys, seekErr = cache.Persist()
 				if seekErr != nil {
+					seekErr = fmt.Errorf("failed to persist intermediate batch of contract storage items and IDs: %w", seekErr)
 					return false
 				}
+				batchCnt++
+				bc.log.Info("intermediate batch of contract storage items and IDs is persisted", zap.Int("batches persisted", batchCnt), zap.Duration("took", time.Since(p)), zap.Int("keys", keys))
+				p = time.Now()
 			}
 			// May safely omit KV copying.
 			k[0] = byte(newStoragePrefix)
 			cache.Store.Put(k, v)
+			cnt++
+			storageItmsCnt++
 
 			// @fixme: remove this part after #2702.
 			if bytes.HasPrefix(k[1:], mgmtContractPrefix) {
 				var hash util.Uint160
 				copy(hash[:], k[mgmtCSPrefixLen:])
 				err = stackitem.DeserializeConvertible(v, cs)
 				if err != nil {
-					seekErr = fmt.Errorf("failed to deserialize contract state: %w", err)
+					bc.log.Warn("failed to deserialize contract; ID for this contract won't be stored in the DB",
```

Comment:
Not a good thing. Luckily it'll be solved with ID->hash maps stored in the contract (#2702), but in general we're losing some data here.

Reply:
These IDs are kind of useless without proper and retrievable contract state inside Management, so I thought that we may act exactly like Management does. But if we need them to be presented anyway, I'll make `DeserializeConvertable` deserialize this contract anyway. Do we?

Reply:
We better not touch deserialization and leave this for #2702 to solve.